mongodbatlas/resource_mongodbatlas_org_invitation.go

```diff
@@ -0,0 +1,269 @@
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error creating Organization invitation: %s for %s", username, err))
+	}
+
+	return nil
+}
+
+func resourceMongoDBAtlasOrgInvitationUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*MongoDBClient).Atlas
+	ids := decodeStateID(d.Id())
+	orgID := ids["org_id"]
+	username := ids["username"]
+	invitationID := ids["invitation_id"]
+
+	InvitationReq := &matlas.Invitation{
+		Roles: createOrgStringListFromSetSchema(d.Get("roles").(*schema.Set)),
+	}
+
+	_, _, err := conn.Organizations.UpdateInvitationByID(ctx, orgID, invitationID, InvitationReq)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error creating Organization invitation: %s for %s", username, err))
+	}
+
+	return resourceMongoDBAtlasOrgInvitationRead(ctx, d, meta)
+}
+
+func resourceMongoDBAtlasOrgInvitationImportState(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	conn := meta.(*MongoDBClient).Atlas
+	orgID, username, err := splitOrgInvitationImportID(d.Id())
+	if err != nil {
+		return nil, err
+	}
+
+	orgInvitations, _, err := conn.Organizations.Invitations(ctx, orgID, nil)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't import Organization invitations, error: %s", err)
+	}
+
+	for _, orgInvitation := range orgInvitations {
+		if orgInvitation.Username != username {
+			continue
+		}
+
+		if err := d.Set("username", orgInvitation.Username); err != nil {
+			return nil, fmt.Errorf("error getting `username` for Organization Invitation (%s): %s", username, err)
+		}
+		if err := d.Set("org_id", orgInvitation.GroupID); err != nil {
+			return nil, fmt.Errorf("error getting `org_id` for Organization Invitation (%s): %s", username, err)
+		}
+		if err := d.Set("invitation_id", orgInvitation.ID); err != nil {
+			return nil, fmt.Errorf("error getting `invitation_id` for Organization Invitation (%s): %s", username, err)
+		}
+		d.SetId(encodeStateID(map[string]string{
+			"username":      username,
+			"org_id":        orgID,
+			"invitation_id": orgInvitation.ID,
+		}))
+		return []*schema.ResourceData{d}, nil
+	}
+
+	return nil, fmt.Errorf("could not import Organization Invitation for %s", d.Id())
+}
+
+func splitOrgInvitationImportID(id string) (orgID, username string, err error) {
+	var re = regexp.MustCompile(`(?s)^([0-9a-fA-F]{24})-(.*)$`)
+	parts := re.FindStringSubmatch(id)
+
+	if len(parts) != 3 {
+		err = fmt.Errorf("import format error: to import a Organization Invitation, use the format {project_id}-{username}")
+		return
+	}
+
+	orgID = parts[1]
+	username = parts[2]
+
+	return
+}
+
+func validateOrgRoles(list *schema.Set) error {
+	if rs := list.List(); list.Len() > 0 {
+		for _, role := range rs {
+			if !validateOrgRole(role.(string)) {
+				return fmt.Errorf("error creating an invite: %s is an invalid role for a Organization", role)
+			}
+		}
+	}
+
+	return nil
```

Comment:
[style] prefer early exits for readability 
```suggestion
	if list.Len() == 0 {
		return nil
	}
	for _, role := range list.List() {
		if !validateOrgRole(role.(string)) {
			return fmt.Errorf("error creating an invite: %s is an invalid role for a Organization", role)
		}
	}
	return nil
```

Reply:
Fixed